spyder/plugins/application/plugin.py

```diff
@@ -156,12 +156,24 @@ def on_mainwindow_visible(self):
             screen.logicalDotsPerInchChanged.connect(
                 container.show_dpi_change_message)
 
-        # Show appeal the fifth time Spyder starts
+        # Show appeal the fifth and 25th time Spyder starts
         spyder_runs = self.get_conf("spyder_runs_for_appeal", default=1)
-        if spyder_runs == 5:
-            container.inapp_appeal_status.show_appeal()
+        if spyder_runs in [5, 25]:
+            # Show in-app appeal message 20 secs after Spyder starts on Mac so
+            # that it appears in front of the main window.
+            # Fixes spyder-ide/spyder#22454
+            if sys.platform == "darwin":
+                QTimer.singleShot(
+                    20000, container.inapp_appeal_status.show_appeal
+                )
+            else:
+                container.inapp_appeal_status.show_appeal()
+
+            # Increase counting in one to not get stuck at this point.
+            # Fixes spyder-ide/spyder#22457
+            self.set_conf("spyder_runs_for_appeal", spyder_runs + 1)
```

Comment:
@ccordoba12 I'm now wondering if, contrary to my original suggestion, rather than a 20s delay for the appeal window we could just ensure that the Application plugin runs `on_mainwindow_visible()` after the `MainWindow.restore_scrollbar_position.emit()`.

```suggestion
        if spyder_runs in [5, 25]:
            container.inapp_appeal_status.show_appeal()

            # Increase counting in one to not get stuck at this point.
            # Fixes spyder-ide/spyder#22457
            self.set_conf("spyder_runs_for_appeal", spyder_runs + 1)
```

Suggestion for `mainwindow.py:850`:
```python
        # Call on_mainwindow_visible for all plugins, except Layout because it
        # needs to be called first (see above) and Application because it nees
        # to be called last (see below).
        for plugin_name in PLUGIN_REGISTRY:
            if plugin_name not in (Plugins.Layout, Plugins.Application):
                plugin = PLUGIN_REGISTRY.get_plugin(plugin_name)
                try:
                    plugin.on_mainwindow_visible()
                    QApplication.processEvents()
                except AttributeError:
                    pass

        self.restore_scrollbar_position.emit()

        # Call Application.on_mainwindow_visible last so that appeal window has
        # focus on macOS. Fixes spyder-ide/spyder#22454.
        PLUGIN_REGISTRY.get_plugin(Plugins.Application).on_mainwindow_visible()
        QApplication.processEvents()
```

Reply:
Sure, I can implement this change if you think it's better. I just have a quick question for you: did you try it locally and it works as expected for you?

Reply:
Yes, I tried it locally and it works.

Reply:
Done.